sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/v2/DataSourceV2Strategy.scala

```diff
@@ -344,13 +344,13 @@ class DataSourceV2Strategy(session: SparkSession) extends Strategy with Predicat
     case DescribeNamespace(ResolvedNamespace(catalog, ns, _), extended, output) =>
       DescribeNamespaceExec(output, catalog.asNamespaceCatalog, ns, extended) :: Nil
 
-    case DescribeRelation(r: ResolvedTable, partitionSpec, isExtended, output) =>
+    case DescribeRelation(r: ResolvedTable, partitionSpec, isExtended, asJson, output) =>
       if (partitionSpec.nonEmpty) {
         throw QueryCompilationErrors.describeDoesNotSupportPartitionForV2TablesError()
       }
       DescribeTableExec(output, r.table, isExtended) :: Nil
```

Comment:
shall we support the AS JSON option in `DescribeTableExec`? If not we should fail explicitly here if `asJson` is true.